File: backend/src/main/java/com/example/backend/services/product/ProductService.java
package com.example.backend.services.product;

import com.example.backend.models.product.ProductRequest;
import com.example.backend.models.product.ProductResponse;

import java.util.List;

public interface ProductService {
    /**
     * Adds a new product to the system.
     *
     * @param productRequest the product to be added
     */
    void addProduct(ProductRequest productRequest);

    /**
     * Updates an existing product.
     *
     * @param productRequest the product with updated information
     */
    void updateProduct(ProductRequest productRequest);

    /**
     * Deletes a product by its ID.
     *
     * @param id the ID of the product to delete
     */
    void deleteProductById(Long id);

    /**
     * Retrieves all products.
     *
     * @return a list of all products
     */
    List<ProductResponse> getAllProducts();

    /**
     * Retrieves a product by its ID.
     *
     * @param id the ID of the product to retrieve
     * @return the product with the given ID
     */
    ProductResponse getProductById(Long id);

    /**
     * Retrieves all products by the owner ID.
     *
     * @param userId the ID of the user to retrieve their products.
     * @return list of products with the given user ID
     */
    List<ProductResponse> getProductsByOwner(Long userId);
}

File: frontend/src/services/ProductService.js
import axiosInstance from "./axiosInstance.js";




export const createProduct = (product) => {
    return axiosInstance.post("/product/create", product)
}

export const updateProduct = (product) => {
    return axiosInstance.put("/product/update", product)
}

export const getAllProducts = () => {
    return axiosInstance.get("/product/all")
}

export const getAllProductsByUserId = (userId) => {
    return axiosInstance.get(`/product/all/${userId}`)
}

export const getProductById = (id) => {
    return axiosInstance.get("/product/" + id)
}

export const getAllProductCategories = () => {
    return axiosInstance.get("/product/categories")
}

export const deleteProduct = (id) => {
    return axiosInstance.delete("/product/delete/" + id)
}